xgi/convert.py

```diff
@@ -146,22 +147,30 @@ def convert_to_line_graph(H):
     ----------
     H : Hypergraph
         The hypergraph of interest
+    s : int
+        The intersection size to consider edges
+        as connected, by default 1.
 
     Returns
     -------
     LG : networkx.Graph
          The line graph associated to the Hypergraph
+
+    References
+    ----------
+    "Hypernetwork science via high-order hypergraph walks"
+    by Sinan G. Aksoy, Cliff Joslyn, Carlos Ortiz Marrero, Brenda Praggastis & Emilie Purvine.
+    https://doi.org/10.1140/epjds/s13688-020-00231-0
     """
 
     LG = nx.Graph()
 
     edge_label_dict = {tuple(edge): index for index, edge in H._edge.items()}
 
-    nodes = sorted(set(edge_label_dict.values()))
-    LG.add_nodes_from(nodes)
+    LG.add_nodes_from(H.edges)
 
     for edge1, edge2 in combinations(H.edges.members(), 2):
-        if edge1.intersection(edge2):
+        if len(edge1.intersection(edge2)) >= s:
             LG.add_edge(edge_label_dict[tuple(edge1)], edge_label_dict[tuple(edge2)])
```

Comment:
It would be faster to collect new edges in a list and then add them with `add_edges_from` I think

Reply:
Good idea!

Reply:
Okay, update is that modifying the for-loop to this:
```
edges_for_adding = []
    for edge1, edge2 in combinations(H.edges.members(), 2):
        if len(edge1.intersection(edge2)) >= s:
            edges_for_adding.append((edge_label_dict[tuple(edge1)], edge_label_dict[tuple(edge2)]))
    LG.add_edges_from(edges_for_adding)
```
is 3% slower than the current algorithm. Let me know if there's a way of writing this that might be faster.

Reply:
Hm okay, unexpected. Then let's leave it as is for now I guess. `add_edges_from` may need some optimizing.